### DISCUSSION TITLE
Can't find how to add new rust files using v2

### DISCUSSION THREAD

**Discussion by clyde-gustafson** [fzyzcjy/flutter_rust_bridge#1892] [Q&A] [answered]
I think this library looks awesome, and the proof of concept added with the codegen is great. 

However, I added a file to my rust/src/api folder and now I can't get the dart side to generate using the codegen. I added the name of the file into lib.rs, and my paths look to be in order in my flutter_rust_bridge.yaml file. Is there a tutorial anywhere on how this is supposed to work?

**Comment by fzyzcjy** [comment]
> However, I added a file to my rust/src/api folder and now I can't get the dart side to generate using the codegen. I added the name of the file into lib.rs, and my paths look to be in order in my flutter_rust_bridge.yaml file. Is there a tutorial anywhere on how this is supposed to work?

Could you please share a minimal reproducible sample (e.g. a github repo)? Without more information, maybe firstly check, do you add `pub mod your_filename;` in `rust/src/api/mod.rs`?

**Reply by clyde-gustafson** [reply]
Yeah I verified that it is in the mod.rs. It's a private repository, but maybe I can see if I can reproduce it on a new flutter app.

**Reply by clyde-gustafson** [reply]
Here we go:

https://github.com/clyde-gustafson/test_rust_bridge

This is after adding the file to my mod.rs and running `flutter_rust_bridge_codegen generate`

**Reply by fzyzcjy** [reply, marked as answer]
Hmm https://github.com/clyde-gustafson/test_rust_bridge/blob/753e78449c8b5b29423fb4ed5d4ea413d69db9b3/rust/src/api/midi_convert.rs#L6 , what about `pub fn`?

**Reply by clyde-gustafson** [reply]
Man that's crappy lol. Thanks for your help. It works now.

**Reply by fzyzcjy** [reply]
Happy to see it work!
